go-controller/pkg/ovn/policy_old.go

```diff
@@ -855,14 +910,19 @@ func (oc *Controller) addNetworkPolicyOld(policy *knet.NetworkPolicy) {
 			if fromJSON.IPBlock != nil {
 				ingress.addIPBlock(fromJSON.IPBlock)
 			}
-			if fromJSON.NamespaceSelector != nil {
+
+			if fromJSON.NamespaceSelector != nil && fromJSON.PodSelector != nil {
+
+				oc.handlePeerNamespaceAndPodSelectorOld(policy,
+					fromJSON.NamespaceSelector, fromJSON.PodSelector,
+					hashedLocalAddressSet, peerPodAddressMap, ingress, np)
+
+			} else if fromJSON.NamespaceSelector != nil {
 				// For each peer namespace selector, we create a watcher that
 				// populates ingress.peerAddressSets
 				oc.handlePeerNamespaceSelectorOld(policy,
 					fromJSON.NamespaceSelector, ingress, np)
-			}
-
-			if fromJSON.PodSelector != nil {
+			} else if fromJSON.PodSelector != nil {
 				// For each peer pod selector, we create a watcher that
 				// populates the addressSet
 				oc.handlePeerPodSelectorOld(policy, fromJSON.PodSelector,
```

Comment:
Don't we also need to do this for egress section?

Reply:
@shettyg added support for egress as well